### PR TITLE
feat: accessing a script that is not running raises an exception

### PR DESCRIPTION
**Pull request by gerard-queralt**
Currently, if a script stops running, you can still get it using `GetComponent`, which might cause an error since the script has no reason to account for that. With this change such cases will raise a separate exception.

### CODE REVIEW

**Review by Piterga12 — DISMISSED** [review]

**Review by markmeliaa — DISMISSED** [review]

**Review by Piterga12 — APPROVED** [review]

**Review by MarcGonzalezMoratona — APPROVED** [review]

File: Source/Exceptions/AxoDefaultException.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+#include <exception>
+
+class AxoDefaultException : public std::exception
+{
+public:
+	AxoDefaultException(const std::string& message);
+	~AxoDefaultException() override = default;
+
+	const char* what() const noexcept override;
```
Comment (MarcGonzalezMoratona) [inline Source/Exceptions/AxoDefaultException.h:11]:
```suggestion
	const char* What() const noexcept override;
```

should not this be What()?

File: Source/DataModels/Components/ComponentScript.cpp [hunk 2]
```diff
@@ -52,6 +53,14 @@ void RunScriptMethodAndHandleException(bool& scriptFailedState, ComponentScript*
 				  exception.what());
 		scriptFailedState = true;
 	}
+	catch (const AccessingFailedScriptException& exception)
+	{
+		LOG_ERROR("Script {}, owned by {} attempted to access script that has stopped running. Error message: {}",
```
Comment (MarcGonzalezMoratona) [inline Source/DataModels/Components/ComponentScript.cpp:58]:
```suggestion
		LOG_ERROR("Script {}, owned by {} attempted to access a script that has stopped running. Error message: {}",
```

Comment (MarcGonzalezMoratona) [reply Source/DataModels/Components/ComponentScript.cpp:58]: you can omit this suggestion if it's too picky!

**Review by MarcGonzalezMoratona — APPROVED** [review]

**Comment by MarcGonzalezMoratona** [comment]
I only checked the code, I suppose you tested it in order to make this functionality

File: Source/Exceptions/AxoDefaultException.h (hunk 1 above)
Comment (gerard-queralt) [reply Source/Exceptions/AxoDefaultException.h:11]: Notice that this is an override, we're extending std::exception: https://en.cppreference.com/w/cpp/error/exception/what

File: Source/DataModels/Components/ComponentScript.cpp (hunk 2 above)
Comment (gerard-queralt) [reply Source/DataModels/Components/ComponentScript.cpp:58]: Picky comments are my speciality :P

**Review by Piterga12 — APPROVED** [review]

File: Source/Exceptions/AxoDefaultException.h (hunk 1 above)
Comment (MarcGonzalezMoratona) [reply Source/Exceptions/AxoDefaultException.h:11]: okay, now I understand it. I didn't see that override. thank you!

**Review by markmeliaa — APPROVED** [review]
LGTM!

**Review by Piterga12 — APPROVED** [review]

**Pull request merged**